src/org/kiwix/kiwixmobile/di/modules/ApplicationModule.java

```diff
@@ -0,0 +1,24 @@
+package org.kiwix.kiwixmobile.di.modules;
+
+import android.app.NotificationManager;
+import android.content.Context;
+import dagger.Module;
+import dagger.Provides;
+import javax.inject.Singleton;
+import org.kiwix.kiwixmobile.KiwixApplication;
+
+@Module public class ApplicationModule {
+  private final KiwixApplication application;
+
+  public ApplicationModule(KiwixApplication application) {
+    this.application = application;
+  }
+
+  @Provides @Singleton Context provideApplicationContext() {
+    return this.application;
+  }
```

Comment:
Shouldn't we add a custom scope @ApplicationScope so we can use it for all services used throughout the app also making the module code more understandable?